csp/sync.csp

```diff
@@ -0,0 +1,187 @@
+        <link rel="stylesheet" type="text/css" href="css/git-webui.css" /> 
+        <server>set $NAMESPACE = "USER"</server>
+        #(##class(SourceControl.Git.Utils).GetSourceControlInclude())#
+        <style type="text/css">
+            h1, h2 {
+                margin-top: 10px
+            }
+
+            .A::after {
+                content: "Added";
+                position: absolute;
+                right: 10px;
+            }
+
+            .D::after {
+                content: "Deleted";
+                position: absolute;
+                right: 10px;
+            }
+
+            .M::after {
+                content: "Modified";
+                position: absolute;
+                right: 10px;
+            }
+
+            .R::after {
+                content: "Renamed";
+                position: absolute;
+                right: 10px;
+            }
+
+            .section-header {
+                margin: 20px 0 15px 0
+            }
+
+            #syncBtn {
+                position: absolute;
+                right: 15px;
+                margin-top: 10px;
+            }
+
+            .output-header {
+                margin-top: 120px;
+            }
+            .output {
+
+                margin-bottom: 20px;
+                border: 1.5px solid darkgray;
+                padding: 10px;
+                border-radius: 4px;
+            }
+
+        </style>
+    </head>
+    <body>
+        <server>
+            set settings = ##class(SourceControl.Git.Settings).%New()
+
+            set uncommittedWithAction = ##class(SourceControl.Git.Utils).UncommittedWithAction().%Get("user")
+            set commitMsg = ##class(SourceControl.Git.Utils).GenerateCommitMessageFromFiles(uncommittedWithAction)
+            set filesToSync = ($length(commitMsg) > 0)
+            set fileSectionDisplay = $select(
+                filesToSync = 1: "block",
+                1: "none"
+            )
+            set noFileDisplay = $select(
+                filesToSync = 0: "block",
+                1: "none"
+            )
+
+            set outputDisplay = "none"
+
+        </server>
+        
+        <div class="container">
+            <h1 class="text-center">Sync Repository</h1>
+            <div class="row">
+                <div class="offset-sm-2 col-sm-8">
+                    <div style="display: #(fileSectionDisplay)#">
+                        <h3 class="section-header">Files to be committed with sync:</h3>
+                        <ul class="list-group">
+                            <server>
+                                set iterator = uncommittedWithAction.%GetIterator()
+                                while iterator.%GetNext(,.uncommitted) {
+                                    set action = uncommitted.%Get("action")
+                                    set file = uncommitted.%Get("file")
+                                    &html<<li class="list-group-item #(action)#">#(file)#</li>>
```

Comment:
Just out of principle let's add protection against cross-site scripting wherever we inject data into the HTML.
```suggestion
                                    &html<<li class="list-group-item #(..EscapeHTML(action))#">#(file)#</li>>
```